bot/exts/moderation/infraction/infractions.py

```diff
@@ -340,6 +344,10 @@ async def apply_ban(
 
         Will also remove the banned user from the Big Brother watch list if applicable.
         """
+        if hasattr(user, 'top_role') and user.top_role >= ctx.me.top_role:
```

Comment:
in which cases do we have a `top_role` here? In case this is a member, right? Would it be cleaner to check for `isinstance` here then? That may help some typecheckers.

Reply:
Yeah, `top_role` is when it's a `discord.Member`. I'm not sure whether I did `hasattr` instead of `isinstance` for a reason or not. I'll check and see - if there's no real reason then `isinstanace` is probably much clearer.

Reply:
Couldn't see a logical reason to stick to `hasattr`, so have pushed a commit changing to `isinstance` :+1:

Thanks for the review :+1: